kotlin-eclipse-ui/src/org/jetbrains/kotlin/wizards/NewProjectWizard.java

```diff
@@ -40,12 +41,18 @@ public boolean performFinish() {
         } catch (InterruptedException e) {
             return false;
         }
-        
-        KotlinNature.Companion.addNature(op.getResult());
-        
-        selectAndRevealResource(op.getResult());
 
-        return true;
+        ProjectCreationOp.OperationResult result = op.getResult();
+        if (result.isSuccess()) {
+            KotlinNature.Companion.addNature(result.getProject());
+            selectAndRevealResource(result.getProject());
+            return true;
+        } else {
+            MessageDialog.openError(getShell(),
+                    MessageDialogContentUtils.getErrorTitleForMessageDialog(result.getException()),
+                    MessageDialogContentUtils.getErrorDescriptionForMessageDialog(result.getException()));
+            return false;
+        }
```

Comment:
Usually, it's better to lift return out of `if` block to avoid fall through:
```
return if (...) {
  ...
 true
} else {
  ....
  false
}
```

Reply:
I am not quite sure you can do it in java...

Reply:
Oh no, I didn't realize it was Java